optimum/onnxruntime/trainer.py

```diff
@@ -260,13 +309,63 @@ def __init__(
             optimizers=optimizers,
             preprocess_logits_for_metrics=preprocess_logits_for_metrics,
         )
-
+        self._training_model = ModuleWithLoss(model, args)
+        self.model = model
+        self._inferencing_model = model
         self.feature = feature
         self.onnx_model_path = onnx_model_path
         self.exported_with_loss = False
         if self.args.local_rank:
             torch.cuda.set_device(self.args.local_rank)
 
+    def _set_signature_columns_if_needed(self):
+        if self._signature_columns is None:
+            # Inspect model forward signature to keep only the arguments it accepts.
+            import inspect
+
+            if self.model == self._training_model:
+                signature = inspect.signature(self.model._original_model.forward)
+            else:
+                signature = inspect.signature(self.model.forward)
+
+            self._signature_columns = list(signature.parameters.keys())
+            # Labels may be named label or label_ids, the default data collator handles that.
+            self._signature_columns += list(set(["label", "label_ids"] + self.label_names))
+
+    def compute_loss(self, model_with_loss, inputs, return_outputs=False):
+        # Run model forward + loss compute.
+        if self.model == self._training_model:
+            outputs = model_with_loss(inputs, return_outputs)
+            return outputs
+        else:
+
+            if self.label_smoother is not None and "labels" in inputs:
```

Comment:
we can just call self.compute_loss(self.model, ...) for the line 340 else branch, right?

Reply:
we can't do that because we override the compute_loss function in ORTTrainer class so we don't have access to the compute_loss function from transformers trainer.

`  File "/home/adamlouly/optimum/optimum/onnxruntime/trainer.py", line 337, in compute_loss
    if self.model == self._training_model:
RecursionError: maximum recursion depth exceeded in comparison`